server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/MailboxAppender.java

```diff
@@ -50,18 +50,13 @@ public MailboxAppender(MailboxManager mailboxManager, Logger logger) {
 
     public void append(MimeMessage mail, String user, String folder) throws MessagingException {
         MailboxSession session = createMailboxSession(user);
-        append(mail, user, folder, session);
+        append(mail, user, useSlashAsSeparator(folder, session), session);
```

Comment:
The path delimiter used for folder hierarchy is defined by the Mailbox session. Hence, someone defining a static folder threw configuration (like INBOX.any) can not be sure it is interpreted as mailbox "any" in mailbox "INBOX", but rather as mailbox with name "INBOX.any" if path separator is not '.'.

Here, we allow to write such a mailbox name as INBOX/any where / will be replaced by the path separator. We make the naming robust to path delimiter changes.